### PR TITLE
Add disk_to_pmtiles: Convert a directory of tiles to pmtiles

### PR DESCRIPTION
**Pull request by larsmaxfield**
- [x] Please do not open PRs for new features if you're a first time contributor; first open an issue to confirm that the new feature is appropriate.

See feature request at https://github.com/protomaps/PMTiles/issues/338.

### CODE REVIEW

**Comment by larsmaxfield** [comment]
Working on fixing the metadata for disk_to_pmtiles.

**Comment by larsmaxfield** [comment]
Fixed.

File: python/pmtiles/convert.py [hunk 1]
```diff
@@ -178,3 +178,167 @@ def pmtiles_to_dir(input, output):
...
+    # Collect a set of all tile IDs
+    z_set = []  # List of all zoom levels for auto-detecting maxzoom.
+    tileid_path_set = []  # List of tile (id, filepath) pairs
+    zoom_dirs = get_dirs(directory_path)
+    zoom_dirs.sort(key=len)
+    try:
+        collect_max = int(maxzoom)
+    except ValueError:
+        collect_max = 99
+    collect_min = metadata.get("minzoom", 0)
```
Comment (bdon) [inline python/pmtiles/convert.py:229]: What tool is creating metadata.json? https://github.com/felt/tippecanoe writes the minzoom and maxzoom as strings, so these may need to be `int(metadata.get...))`

example tippecanoe command: `./tippecanoe --output-to-directory foo ne_10m_admin_0_label_points.fgb --maximum-zoom 5 --force`

Comment (bdon) [reply python/pmtiles/convert.py:229]: tippecanoe also writes a key called `json` that has `vector_layers`. I think this code will need to special-case `format=pbf` in order to translate `vector_layers` correctly as they are required in the spec:

https://github.com/protomaps/PMTiles/blob/main/spec/v3/spec.md#5-json-metadata

The other key it writes is `tilestats` - so all top-level keys of the `json` value could lifted up one level in the pmtiles metadata.

Tools other than tippecanoe might write different `metadata.json`?

Comment (bdon) [reply python/pmtiles/convert.py:229]: If the above sounds like overkill then maybe we limit the disk_to_pmtiles to raster formats only (if that's the use case you need)

Comment (larsmaxfield) [reply python/pmtiles/convert.py:229]: Ah I wasn't aware of that. Thanks for the explanation. 

For now let's limit it to raster formats. That's indeed the only use case I need. If someone with a need (and knowledge) for other formats wishes to add support, there's at least this function for them to add to.

Comment (larsmaxfield) [reply python/pmtiles/convert.py:229]: Are you OK if I simply remove 'pbf` from the help and docstrings of the functions?

Or would you like an explicit rejection for formats not 'png', 'jpeg', 'webp', or 'avif'?

Comment (bdon) [reply python/pmtiles/convert.py:229]: removing `pbf` is fine.

Comment (larsmaxfield) [reply python/pmtiles/convert.py:229]: References to 'pbf' are now gone, and I added 'raster format' to be more specific.

**Comment by bdon** [comment]
3.4.0 is published to https://pypi.org/project/pmtiles/ now, thanks!

**Comment by larsmaxfield** [comment]
Nice and efficient! Happy to help.

**Pull request merged**